### DISCUSSION TITLE
How to pass custom value using string literal? `w-[${CardSize.XS.width}]`

### DISCUSSION THREAD

**Discussion by RodrigoTomeES** [tailwindlabs/tailwindcss#7774] [Help] [answered]
Hi,

I am having trouble when use string literal, Tailwind instead of evaluate it and later pass the value, it pass the string literal, there is any workaround?

![imagen](https://user-images.githubusercontent.com/13077343/157004926-c916b683-84cd-4f15-b566-36847e6fe77d.png)

**Comment by mflorida** [comment, marked as answer]
This is a known limitation. The suggested solutions are to use the full text of the class you want to use somewhere in your component or use Tailwind's `@apply` directive. You could set up an object that maps the possible widths/heights to the full class string. This example uses React, but the concept would apply to any front-end framework.

```javascript
// define an object mapping the 'sizes' to the class strings
const cardSizes = {
  // use the full text of desired Tailwind CSS classes
  XS: 'w-8 h-4',
  S: 'w-16 h-8'
}
// use as an argument for your template renderer
function Template({ size })
  X.args = {
    style: {},
    className: `${cardSizes[size]}`
  }
  return <div className={X.args.className}>Hello</div>
}
// pass size string when calling the template function
<Template size={'XS'}/>
```
Or use Tailwind's `@apply` directive in a CSS file:
```css
.card-xs: {
  @apply w-8 h-4;
}
.card-sm: {
  @apply w-16 h-8;
}
```

**Reply by RodrigoTomeES** [reply]
Great, I will try it tomorrow. Thanks! 🚀
